src/components/dialogs/dialogAlert/dialogAlertHeader/dialogAlertHeader.tsx

```diff
@@ -0,0 +1,55 @@
+import { Title } from '@radix-ui/react-alert-dialog';
+import classNames from 'classnames';
+import type React from 'react';
+import { useContext, type ComponentPropsWithoutRef } from 'react';
+import { AvatarIcon } from '../../../avatars';
+import { IconType } from '../../../icon';
+import { DialogAlertContext, type DialogAlertVariant } from '../dialogAlertRoot';
+
+export interface IDialogAlertHeaderProps extends ComponentPropsWithoutRef<'div'> {
+    /**
+     * Title summarizing dialog's content or purpose.
+     */
+    title: string;
+}
+
+const dialogAlertVariantToIcon: Record<DialogAlertVariant, IconType> = {
+    critical: IconType.CRITICAL,
+    info: IconType.INFO,
+    success: IconType.SUCCESS,
+    warning: IconType.WARNING,
+};
+
+const dialogAlertVariantToTitleClass: Record<DialogAlertVariant, string> = {
+    critical: 'text-critical-600',
+    info: 'text-info-600',
+    success: 'text-success-600',
+    warning: 'text-warning-600',
+};
+
+/**
+ * `DialogAlert.Header` component
+ *
+ * **NOTE**: This component must be used inside a `<DialogAlert.Root />` component.
+ */
+export const DialogAlertHeader: React.FC<IDialogAlertHeaderProps> = ({ title }) => {
+    const { variant } = useContext(DialogAlertContext);
```

Comment:
We need to pass all the `otherProps` object to the main div 
```suggestion
export const DialogAlertHeader: React.FC<IDialogAlertHeaderProps> = (props) => {
    const { title, ...otherProps } = props;
    const { variant } = useContext(DialogAlertContext);
```